x-pack/test/accessibility/apps/automated.ts

```
@@ -0,0 +1,90 @@
+}
+
+function readErrors() {
+  let yaml;
+  try {
+    yaml = Fs.readFileSync(PATH, 'utf8');
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      throw error;
+    }
+  }
+
+  return yaml ? Yaml.safeLoad(yaml) : {};
+}
+
+function updateErrors(originalErrors: Errors, currentErrors: Errors) {
+  const updatedErrors: Errors = {};
+
+  for (const [key, value] of Object.entries(currentErrors)) {
+    const originalCount = originalErrors[key];
+
+    if (originalCount && originalCount < value) {
+      updatedErrors[key] = originalCount;
+    } else {
+      updatedErrors[key] = value;
+    }
+  }
+
+  Fs.writeFileSync(PATH, Yaml.safeDump({ errorsPerPage: updatedErrors }));
+}
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const { common, header } = getPageObjects(['common', 'header']);
+  const a11y = getService('a11y');
+  const config = getService('config');
+  const log = getService('log');
+
+  let originalErrors = {};
+  const currentErrors: Errors = {};
+
+  describe('Automated pass', () => {
+    before(async () => {
+      await common.navigateToActualUrl('home');
+      originalErrors = readErrors();
+    });
+    after(() => {
+      updateErrors(originalErrors, currentErrors);
+    });
+
+    it('runs', async () => {
+      const apps: Array<[string, { pathname: string; hash?: string }]> = Array.from(
+        Object.entries(config.get('apps'))
+      );
+
+      for (const [app, { hash }] of apps) {
+        if (app === 'login') continue; // causes a perpetual login loop, ignore
+        await common.navigateToActualUrl(app, hash);
+        await header.waitUntilLoadingHasFinished();
+        const errorCount = await a11y.testAppSnapshot({
+          returnErrorCount: true,
+          reportType: 'all',
+        });
+        if ((errorCount ?? 0) > (currentErrors[app] ?? 0)) {
+          log.debug(
+            `App: ${app}; Errors: ${errorCount ?? 0}; Max allowed: ${currentErrors[app] ?? 0}`
+          );
```

This is the info that's helpful in debugging but it doesn't seem like this gets printed to the console in CI. What's the best way for me to include this all in the error?

---

I'd just throw an error with that as the message, that's all the `expect(false).to.be.ok()` is doing.